Source/WebCore/platform/audio/cocoa/AudioSampleBufferList.cpp

```diff
@@ -112,40 +110,42 @@ void AudioSampleBufferList::applyGain(float gain)
 
 static void mixBuffers(WebAudioBufferList& destinationBuffer, const AudioBufferList& sourceBuffer, AudioStreamDescription::PCMFormat format, size_t frameCount)
 {
-    for (uint32_t i = 0; i < destinationBuffer.bufferCount(); i++) {
+    auto sourceBufferSpan = span(sourceBuffer);
+    auto destinationBufferSpan = span(*destinationBuffer.list());
+    for (auto [source, destination] : zippedRange(sourceBufferSpan, destinationBufferSpan)) {
         switch (format) {
         case AudioStreamDescription::Int16: {
-            ASSERT(frameCount <= destinationBuffer.buffer(i)->mDataByteSize / 2);
-            ASSERT(frameCount <= sourceBuffer.mBuffers[i].mDataByteSize / 2);
+            ASSERT(frameCount <= destination.mDataByteSize / 2);
+            ASSERT(frameCount <= source.mDataByteSize / 2);
 
-            auto* destination = static_cast<int16_t*>(destinationBuffer.buffer(i)->mData);
-            auto* source = static_cast<int16_t*>(sourceBuffer.mBuffers[i].mData);
-            for (size_t i = 0; i < frameCount; i++)
-                destination[i] += source[i];
+            auto sourceData = dataSpan<int16_t>(source);
+            auto destinationData = dataMutableSpan<int16_t>(destination);
+            for (auto [s, d] : zippedRange(sourceData, destinationData))
+                d += s;
             break;
         }
         case AudioStreamDescription::Int32: {
-            ASSERT(frameCount <= destinationBuffer.buffer(i)->mDataByteSize / 4);
-            ASSERT(frameCount <= sourceBuffer.mBuffers[i].mDataByteSize / 4);
+            ASSERT(frameCount <= destination.mDataByteSize / 4);
```

Comment:
I think `4` would read more clearly as `sizeof(int32_t)`. (I think that's what we're trying to match here?)